src/lib.rs

```diff
@@ -176,14 +177,22 @@ pub use poll::Poll;
 // Primitive futures
 mod done;
 mod empty;
+#[cfg(not(feature = "never"))]
 mod failed;
+#[cfg(not(feature = "never"))]
```

Comment:
Ah yeah I was thinking we'd either avoid adding those for now but we shouldn't change the existing types as then enabling `never` would be a breaking change for the crate.